### PR TITLE
Integration Testing 2

### PR DESCRIPTION
**Pull request by khushi2305**
This PR tends to cover all possible flows present in the app - 
1 ) When direct payment is done after choosing policy .
2) When no valid offer is available and the user goes for payment without any offer
3) Checked the routing  for other options in menu bar such contact us and viewing smart device discounts .

Also , added a widget test in payment page for case of "No discount chosen". Added a App Bar to Contact Us Page (since flutter driver couldn't use system backs while testing  ).
All the test cases passed .

### CODE REVIEW

**Comment by khushi2305** [comment]
@googlebot I consent.

File: home_insurance_app/test_driver/app_test.dart [hunk 1]
```diff
@@ -76,24 +114,23 @@ dynamic choosePolicy() async {
...
+      find.text("Avail Smart Device Discounts");
   await flutterDriver.tap(viewDiscountsButton);
 
   await flutterDriver.waitFor(find.byType("DisplayDiscounts"));
 }
 
 // Click on Payment on Policy Page
 dynamic paymentAfterPolicy() async {
-  SerializableFinder paymentButton = find.text("Payment");
+  SerializableFinder paymentButton = find.text("Skip to Payment");
```
Comment (dreamplay) [inline home_insurance_app/test_driver/app_test.dart:126]: can we get rid of text

File: home_insurance_app/test_driver/app_test.dart [hunk 2]
```diff
@@ -76,24 +114,23 @@ dynamic choosePolicy() async {
...
   await flutterDriver.tap(paymentButton);
-
   await flutterDriver.waitFor(find.byType("Payment"));
-
 }
 
 // Click on add devices on Show Discounts page
 dynamic addDevices() async {
-  SerializableFinder addDevicesButton = find.text("Add Devices");
+  SerializableFinder addDevicesButton = find.text("Link Devices");
```
Comment (dreamplay) [inline home_insurance_app/test_driver/app_test.dart:133]: can we get rid of text

File: home_insurance_app/test_driver/app_test.dart [hunk 3]
```diff
@@ -119,9 +155,16 @@ dynamic selectOffer() async {
 //      await Future.delayed(const Duration(seconds: 1));
 }
 
+dynamic paymentWithNoOfferSelected() async {
+  SerializableFinder submitButton = find.text("Go to Payment");
```
Comment (dreamplay) [inline home_insurance_app/test_driver/app_test.dart:159]: can we get rid of text

**Comment by khushi2305** [comment]
@googlebot I consent.

**Comment by anjalisoni3655** [comment]
@googlebot I consent.

**Pull request merged**